db/migrate/20210115124508_template_kind_registration.rb

```diff
@@ -0,0 +1,49 @@
+class TemplateKindRegistration < ActiveRecord::Migration[6.0]
+  def up
+    host_init_config_kind = TemplateKind.unscoped.find_or_create_by(name: 'host_init_config')
+    registration_kind = TemplateKind.unscoped.find_by(name: 'registration')
+
+    # Migrate Host registration templates to host_init_config kind
+    ProvisioningTemplate.unscoped
+                        .where(template_kind: registration_kind)
+                        .where.not(name: [Setting[:default_global_registration_item], 'Global Registration'])
+                        .update_all(template_kind_id: host_init_config_kind.id)
+
+    # Rename 'Linux registration default' to 'Linux host_init_config default'
+    ProvisioningTemplate.unscoped
+                        .where(name: 'Linux registration default')
+                        .update_all(name: Setting[:default_host_init_config_template])
+
+    # Assign default host_init_config template to all operating systems
+    # and change registration association to the host_init_config
+    template = ProvisioningTemplate.unscoped.find_by_name(Setting[:default_host_init_config_template])
```

Comment:
The template `Linux registration default`  might not have existed yet or the name could be changed at the time of migration and thus `template` may be nil here. Can we prevent migrations failing for such cases?

Reply:
cc @stejskalleos

Reply:
Fix: https://github.com/theforeman/foreman/pull/9160